device.yml

```diff
@@ -0,0 +1,116 @@
+%YAML 1.1
+---
+# yaml-language-server: $schema=https://raw.githubusercontent.com/harp-tech/reflex-generator/main/schema/device.json
+device: InputExpander
+whoAmI: 1106
+firmwareVersion: "2.2"
+hardwareTargets: "1.2"
+registers:
+  AuxInPort:
+    address: 32
+    type: U8
+    access: Event
+    maskType: AuxiliaryInputs
+    description: Reports the state of the auxiliary inputs.
+  AuxInRisingEdge:
+    address: 33
+    type: U8
+    access: Write
+    maskType: AuxiliaryInputs
+    description: Enables rising edge detection on the auxiliary inputs.
+  AuxInFallingEdge:
+    address: 34
+    type: U8
+    access: Write
+    maskType: AuxiliaryInputs
+    description: Enables falling edge detection on the auxiliary input port.
+  DigitalInput:
+    address: 35
+    type: U16
+    access: Event
+    length: 2
+    payloadSpec:
+      DigitalInputState:
+        offset: 0
+        maskType: DigitalInputs
+        description: Reports the state of all digital inputs in the port.
+      DigitalInputChanged:
+        offset: 1
+        maskType: DigitalInputs
+        description: Reports which digital inputs changed state in the port.
+    description: Reports the state of the digital inputs.
+  DigitalInputEnableRisingEdge:
+    address: 36
+    type: U16
+    access: Write
+    maskType: DigitalInputs
+    description: Enables rising edge detection on the digital input port.
+  DigitalInputFallingEdge:
+    address: 37
+    type: U16
+    access: Write
+    maskType: DigitalInputs
+    description: Enables falling edge detection on the digital input port.
+  InputSamplingMode:
+    address: 38
+    type: U8
+    access: Write
+    maskType: InputSamplingConfig
+    description: Configures the input sampling mode.
+  EncoderSamplingMode:
```

Comment:
"sampling" or "sample" is another question that keeps coming up. In the output expander we went with "sample" but not sure if that is what we agreed in the end.

Reply:
Yes, we had this discussion, my bias is still to use sampling not sure why, it just "sounds" better but we all agreed on sample.